### DISCUSSION TITLE
Next/Image - MaxDimensions for the image component

### DISCUSSION THREAD

**Discussion by thadeucity** [vercel/next.js#28006] [Ideas]
### Describe the feature you'd like to request

Support for **max-width** and **max-heigh** when using `layout: fill` to prevent unnecessary bandwidth waste. 

Example:
When I know for sure that my image will never extrapolate the 80px from my container width I don't need to load an image larger than that nor I need Vercel to create an 500px wide version of this image. 

### Describe the solution you'd like

```jsx
// Original image size (2000px)

<div className="image-container" style={{position:"relative", width: '5rem', height: '5rem'}}>
  <Image
    src={randomImage}
    alt="test image"
    layout="fill"
    maxWidth={160} <<<<<<<<<<<
    maxHeight={160} <<<<<<<<<<<
    objectFit="cover"
  />
</div>
```

With my max dimensions being set to 2x my default 5 REM value I guarantee that my image will load nicely for users with huge default font-sizes. And at the same time I will cap the max dimensions for my imagem set to prevent unnecessary waste of bandwidth

### Describe alternatives you've considered

An option to fit the fixed size imagem inside its container without overflowing it if you resize the container.

example:
(This example will cause the image to overflow or be trimmed by its container for smaller viewports)
```jsx
css`
@media (max-width: 720px) {
  html {
    font-size: 87.5%;
  }
}
`

<div className="image-container" style={{position:"relative", width: '5rem', height: '5rem'}}>
  <Image
    src={randomImage}
    alt="test image"
    layout="fixed"
    width={80}
    height={80}
    objectFit="cover"
    adjustToContainer={true} <<< not a great name but this is the idea
  />
</div>
```

**Comment by lveillard** [comment]
Feel free to check this discussion https://github.com/vercel/next.js/discussions/18312
And this feature request that would solve this also: 
https://github.com/vercel/next.js/discussions/29340
